wsmaster/che-core-api-factory-azure-devops/src/main/java/org/eclipse/che/api/factory/server/azure/devops/AzureDevOpsPersonalAccessTokenFetcher.java

```diff
@@ -172,8 +172,19 @@ public Optional<Boolean> isValid(PersonalAccessToken personalAccessToken) {
   public Optional<Pair<Boolean, String>> isValid(PersonalAccessTokenParams params)
       throws ScmCommunicationException {
     if (!isValidScmServerUrl(params.getScmProviderUrl())) {
-      LOG.debug("not a valid url {} for current fetcher ", params.getScmProviderUrl());
-      return Optional.empty();
+      if (OAUTH_PROVIDER_NAME.equals(params.getScmProviderName())) {
```

Comment:
Can we refactor this fuction?
For instance, move `if (OAUTH_PROVIDER_NAME.equals(params.getScmProviderName())) {`  before ` if (!isValidScmServerUrl(params.getScmProviderUrl())) {`

Reply:
We need to check if the token belongs to Azure DevOps SAAS or Server first. The `isValidScmServerUrl` function name is a bit misleading, so I renamed it.